src/web/pages/portlists/PortListComponent.jsx

```diff
@@ -183,7 +183,18 @@ class PortListComponent extends React.Component {
     return Promise.all(promises)
       .then(() => save(data))
       .then(() => this.closePortListDialog())
-      .catch(error => this.props.onSaveError(error));
+      .catch(error => {
+        if (data.id) {
+          if (isDefined(this.props.onSaveError)) {
+            return this.props.onSaveError(error);
+          }
+        } else {
+          if (isDefined(this.props.onCreateError)) {
+            return this.props.onCreateError(error);
+          }
+        }
+        return Promise.reject(error);
+      });
```

Comment:
Q: could we reduce it to something like this? 
In any case, L:191 could become `else if`

```suggestion
      .catch(error => {
        if (data.id && isDefined(this.props.onSaveError)) {
          return this.props.onSaveError(error);
        } else if (!data.id && isDefined(this.props.onCreateError)) {
          return this.props.onCreateError(error);
        }
        return Promise.reject(error);
      });
```

Reply:
Of course